src/Miningcore/Blockchain/Ethereum/EthereumPayoutHandler.cs

```diff
@@ -890,24 +885,30 @@ private async Task PayoutBalancesOverThresholdAsync()
         {
             logger.Info(() => $"[{LogCategory}] Processing payout for pool [{poolConfig.Id}]");
 
-            var poolBalancesOverMinimum = await TelemetryUtil.TrackDependency(() => cf.Run(con =>
-                    balanceRepo.GetPoolBalancesOverThresholdAsync(con, poolConfig.Id, poolConfig.PaymentProcessing.MinimumPayment)),
-                DependencyType.Sql, "GetPoolBalancesOverThresholdAsync", "GetPoolBalancesOverThresholdAsync");
-
-            if(poolBalancesOverMinimum.Length > 0)
+            try
             {
-                try
+                var minimumPayout = await GetMinimumPayout();
+
+                // Get the balances over the dynamically calculated threshold
+                var poolBalancesOverMinimum = await TelemetryUtil.TrackDependency(() => cf.Run(con =>
+                        balanceRepo.GetPoolBalancesOverThresholdAsync(con, poolConfig.Id, minimumPayout)),
```

Comment:
this Query returns an unlimited number of rows.  We are only going to use {batchSize} of them.

wouldn't it be more efficient to pass batchsize into this method, and add "LIMIT @batchSize" to the query?